test/Microsoft.ML.Tests/Transformers/PcaTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Data;
+using Microsoft.ML.Runtime.Data.IO;
+using Microsoft.ML.Runtime.RunTests;
+using Microsoft.ML.Transforms;
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Microsoft.ML.Tests.Transformers
+{
+    public sealed class PcaTests : TestDataPipeBase
+    {
+        public PcaTests(ITestOutputHelper helper)
+            : base(helper)
+        {
+        }
+
+        [Fact]
+        public void TestPca()
+        {
+            var env = new ConsoleEnvironment(seed: 1, conc: 1);
+            string dataSource = GetDataPath("generated_regression_dataset.csv");
+            var data = TextLoader.CreateReader(env,
+                c => (label: c.LoadFloat(11), features: c.LoadFloat(0, 10)),
+                separator: ';', hasHeader: true)
+                .Read(new MultiFileSource(dataSource));
+
+            var invalidData = TextLoader.CreateReader(env,
+                c => (label: c.LoadFloat(11), features: c.LoadText(0, 10)),
+                separator: ';', hasHeader: true)
+                .Read(new MultiFileSource(dataSource));
+
+            var est = new PcaEstimator(env, "features", "pca", null, rank: 5, seed: 1);
+            //TestEstimatorCore(est, data.AsDynamic, invalidInput: invalidData.AsDynamic);
+
```

Comment:
?

Reply:
failing on empty data view. Added description and link to issue.

---
In reply to: [220037180](https://github.com/dotnet/machinelearning/pull/1017#discussion_r220037180) [](ancestors = 220037180)